### DISCUSSION TITLE
problem with lcd and encoder

### DISCUSSION THREAD

**Discussion by tizfra** [probonopd/MiniDexed#130] [Q&A] [answered]
Hi everyone,
i have connected all in the right way, but lcd starts write character after one column, so i can read 
▯▯▯▯▯▯▯▯Minidexe
▯<TG2▯▯▯▯▯▯▯▯▯▯▯

the encoder problem is that raspberry don't read push button well, it missed lot of push.
can anyone help me

**Comment by probonopd** [comment]
Hello @tizfra. Maybe loose connections? Did you solder?

**Reply by tizfra** [reply]
This could be for the encoder problem, but the LCD problem seems like raspberry starts counting character from 1 and LCD starts from 0...

**Reply by probonopd** [reply]
This could also be caused by imperfect connections. As far as I remember I had this during testing sometimes.

**Reply by tizfra** [reply]
I've also used level converter between raspberry and LCD because LCD is 5v and raspberry is 3.3v

**Reply by probonopd** [reply]
Maybe that is why it is not working properly? I am not using a level converter. Since the connections only go from the Pi to the LCD (and not the other way around) I _thin_ this should be safe, but no guarantees :)

**Reply by tizfra** [reply]
Have you power LCD with 5v?
If possible, in the readme file, add electric scheme...

**Reply by probonopd** [reply]
Yes. We might still want to change the pinout, hence it is not documented in detail yet.

**Reply by arsamus** [reply]
I have some doubts about connecting the LCD. If the LCD is powered by 5v line it is supposed that never works as Read mode to avoid 5v over 3,3v GPIO lines, however I see on schematics that RW is connected to GPIO16 and not to GND. Is the software reads the LCD? or GPIO16 is always set on "0"? or LCD has to be powered by 3.3 line?

**Reply by probonopd** [reply]
Please have a look at [this diagram](https://github.com/probonopd/MiniDexed/issues/38#issuecomment-1062069267).   Note that we might still change the default pinout.

**Reply by arsamus** [reply]
Ok, Thanks. I've got the answers on that comments [issue comment](https://github.com/probonopd/MiniDexed/issues/38#issuecomment-1062155490). 
Maybe it's better to connect LCD RW pin to GND, because, as it's said "The Circle LCD driver is not reading from the display"  

> Please have a look at [this diagram](https://github.com/probonopd/MiniDexed/issues/38#issuecomment-1062069267). Note that we might still change the default pinout.

**Comment by luisfcorreia** [comment]
to work properly, the encoder needs resistors and capacitors, like norns sheild had
(I could work on that, tomorrow)

**Reply by probonopd** [reply]
My encoder module PCB has some resistors onboard. Which capacitor is needed?

**Comment by tizfra** [comment, marked as answer]
I've resolved my problems with encoder and LCD, i don't connected well and on breadboard and I remove also level converter.
Now it works like a charm.

Can I add more voice in micro SD?
Thanks to everyone for the help

**Reply by probonopd** [reply]
Sure you can! Just name them like the examples in https://github.com/probonopd/MiniDexed/blob/main/getsysex.sh:

* sysex/voice/000000_YOURNAME.syx
* sysex/voice/000001_YOURNAME.syx
* sysex/voice/000002_YOURNAME.syx
* ...
